pingora-core/src/listeners/mod.rs

```diff
@@ -118,6 +119,13 @@ impl UninitializedStream {
             Ok(Box::new(self.l4))
         }
     }
+
+    /// Get the peer address of the connection if available
+    pub fn peer_addr(&self) -> Option<String> {
```

Comment:
```suggestion
    pub fn peer_addr(&self) -> Option<SocketAddr> {
        self.l4
            .get_socket_digest()
            .and_then(|d| d.peer_addr().cloned())
    }
```
It actually looks like we can defer the conversion to String, and instead pass around the `l4::socket::SocketAddr` struct.